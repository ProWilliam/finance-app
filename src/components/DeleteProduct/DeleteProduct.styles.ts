import { StyleSheet } from 'react-native';

const StylesDeleteProduct = StyleSheet.create({
  overlay: {
    flex: 1,
    justifyContent: 'flex-end',
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  alertContainer: {
    backgroundColor: 'white',
    padding: 50,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    alignItems: 'center',
  },
  alertTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 20,
  },
  buttonContainer: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'space-between',
    width: '100%',
  },  
});

export default StylesDeleteProduct;