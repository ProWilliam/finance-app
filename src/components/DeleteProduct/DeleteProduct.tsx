import React from 'react';
import { View, Text, TouchableOpacity, Modal } from 'react-native';

import styles from './DeleteProduct.styles';
import ButtonSubmit from '../Button/ButtonSubmit';
import DeletedProductProps from '../../types/components/DeletedProduct.types';
import config from '../../../app.config';

const DeleteProduct: React.FC<DeletedProductProps> = ({ visible, onConfirm, onCancel, info, id }) => {

  return (
    <Modal
      transparent={true}
      animationType="slide"
      visible={visible}
      onRequestClose={onCancel}
    >
      <View style={styles.overlay}>
        <View style={styles.alertContainer}>
          <Text style={styles.alertTitle}>{`¿Estás seguro de eliminar el producto? \nNombre: ${info} \nId: ${id} ?`}</Text>
          <View style={styles.buttonContainer}>
            <ButtonSubmit title='Eliminar' color='deleted' press={onConfirm} navigationRoot={config.extra.rootHome}></ButtonSubmit>
            <ButtonSubmit title='Cancelar' color='edit' press={onCancel}></ButtonSubmit>
          </View>
        </View>
      </View>
    </Modal>
  );
};

export default DeleteProduct;