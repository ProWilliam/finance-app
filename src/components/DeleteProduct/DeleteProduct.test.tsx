import React from 'react';
import renderer, { act } from 'react-test-renderer';

import DeleteProduct from './DeleteProduct';
import styles from './DeleteProduct.styles';

// Mock useAppNavigation hook
const mockAppNavigation = jest.fn();

jest.mock('../../hooks/useAppNavigation', () => ({
  __esModule: true,
  default: () => ({
    appNavigation: mockAppNavigation,
  }),
}));

describe('<DeleteProduct />', () => {

  const mockOnConfirm = jest.fn();
  const mockOnCancel = jest.fn();
  const testInfo = 'Producto de prueba 1';
  const testId = '1';

  it('should render correctly when visible', () => {
    const tree = renderer.create(
      <DeleteProduct 
        visible={true}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
        info={testInfo}
        id={testId}
      />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
  });

  it('should call onConfirm when "Eliminar" is pressed', () => {
    const component = renderer.create(
      <DeleteProduct 
        visible={true}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
        info={testInfo}
        id={testId}
      />
    );
    
    const eliminarButton = component.root.findByProps({ title: 'Eliminar' }).props;
    
    act(() => {
      eliminarButton.press();
    });

    expect(mockOnConfirm).toHaveBeenCalled();
  });

  it('should call onCancel when "Cancelar" is pressed', () => {
    const component = renderer.create(
      <DeleteProduct 
        visible={true}
        onConfirm={mockOnConfirm}
        onCancel={mockOnCancel}
        info={testInfo}
        id={testId}
      />
    );
    
    const cancelarButton = component.root.findByProps({ title: 'Cancelar' }).props;
    
    act(() => {
      cancelarButton.press();
    });

    expect(mockOnCancel).toHaveBeenCalled();
  });

  it('should display the correct product name and ID', () => {
    
    const tree = renderer.create(
      <DeleteProduct 
        visible={true} 
        onConfirm={mockOnConfirm} 
        onCancel={mockOnCancel} 
        info={testInfo} 
        id={testId} 
      />
    );

    const instance = tree.root;
    const textElement = instance.findByProps({ style: styles.alertTitle }).props.children;

    expect(textElement).toContain(`Nombre: ${testInfo}`);
    expect(textElement).toContain(`Id: ${testId}`);
  });

  it('should call onConfirm when the "Eliminar" button is pressed', () => {
    
    const tree = renderer.create(
      <DeleteProduct 
        visible={true} 
        onConfirm={mockOnConfirm} 
        onCancel={jest.fn()} 
        info='Producto de prueba' 
        id='123' 
      />
    );
    const eliminarButton = tree.root.findByProps({ title: 'Eliminar' }).props;

    act(() => {
      eliminarButton.press();
    });
    
    expect(mockOnConfirm).toHaveBeenCalled();
  });
  
  it('should call onCancel when the "Cancelar" button is pressed', () => {
    
    const tree = renderer.create(
      <DeleteProduct 
        visible={true} 
        onConfirm={jest.fn()} 
        onCancel={mockOnCancel} 
        info='Producto de prueba' 
        id='123' 
      />
    );
    const cancelarButton = tree.root.findByProps({ title: 'Cancelar' }).props;

    act(() => {
      cancelarButton.press();
    });
    
    expect(mockOnCancel).toHaveBeenCalled();
  });

});