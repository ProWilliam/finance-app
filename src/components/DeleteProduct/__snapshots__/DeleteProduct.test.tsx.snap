// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteProduct /> should render correctly when visible 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "padding": 50,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 20,
          }
        }
      >
        ¿Estás seguro de eliminar el producto? 
Nombre: Producto de prueba 1 
Id: 1 ?
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F70000",
              "borderRadius": 10,
              "height": 60,
              "justifyContent": "center",
              "marginVertical": 10,
              "opacity": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 10,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#1E2126",
                "fontSize": 14,
                "fontWeight": "bold",
                "padding": 10,
              }
            }
          >
            Eliminar
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#eaeaea",
              "borderRadius": 10,
              "height": 60,
              "justifyContent": "center",
              "marginVertical": 10,
              "opacity": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 10,
              "width": "80%",
            }
          }
        >
          <Text
            style={
              {
                "color": "#1E2126",
                "fontSize": 14,
                "fontWeight": "bold",
                "padding": 10,
              }
            }
          >
            Cancelar
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
